include/libtorrent/session_handle.hpp

```diff
@@ -449,6 +449,14 @@ namespace libtorrent
 		// posted, regardless of the alert mask.
 		void dht_live_nodes(sha1_hash const& nid);
 
+		// Query the DHT to retrieve a sample of the infohashes that the specific
+		// node currently have in their storage.
+		// The ``target`` is included for iterative lookups so that indexing nodes
+		// can perform a keyspace traversal with a single RPC per node by adjusting
+		// the target value for each RPC. It has no effect on the returned sample value.
+		// The result is posted as a ``dht_sample_infohashes_alert``.
+		void dht_sample_infohashes(udp::endpoint const& ep, sha1_hash const& target);
```

Comment:
Indexing the DHT will involve traversing many times more nodes than libtorrent keeps in its routing table. The client would only use `dht_live_nodes` to get the initial batch of nodes to query. The list of `nodes` returned for the query needs to be included in `dht_sample_infohashes_alert` so that the client can implement its own traversal algorithm.